python/cudf/cudf/core/index.py

```diff
@@ -428,8 +433,76 @@ def __array_function__(self, func, types, args, kwargs):
         else:
             return NotImplemented
 
-    def isin(self, values):
-        return self.to_series().isin(values)
+    def isin(self, values, level=None):
+        """Return a boolean array where the index values are in values.
+
+        Compute boolean array of whether each index value is found in
+        the passed set of values. The length of the returned boolean
+        array matches the length of the index.
+
+        Parameters
+        ----------
+        values : set, list-like, Index or Multi-Index
+            Sought values.
+        level : str or int, optional
+            Name or position of the index level to use (if the index
+            is a MultiIndex).
+        Returns
+        -------
+        is_contained : cupy array
+            CuPy array of boolean values.
+
+        """
+
+        if isinstance(self, cudf.MultiIndex):
+            if level is None:
+                if isinstance(values, cudf.MultiIndex):
+                    values_idx = values
+                elif (
+                    (
+                        isinstance(
+                            values,
+                            (
+                                cudf.Series,
+                                cudf.Index,
+                                cudf.DataFrame,
+                                column.ColumnBase,
+                            ),
+                        )
+                    )
+                    or (not is_list_like(values))
+                    or (
+                        is_list_like(values)
+                        and len(values) > 0
+                        and not isinstance(values[0], tuple)
+                    )
+                ):
+                    raise TypeError(
+                        "values need to be a list of tuple squences."
```

Comment:
This TypeError should be more descriptive that input can be `set, list-like, Index or Multi-Index`

Reply:
Decoupled the `Index.isin` to `MultiIndex.isin` and `Index.isin`.